addon/-legacy-private/system/relationships/state/has-many.js

```diff
@@ -249,13 +264,14 @@ export default class ManyRelationship extends Relationship {
   reload() {
     let manyArray = this.manyArray;
 
-    if (this._loadingPromise) {
-      if (this._loadingPromise.get('isPending')) {
-        return this._loadingPromise;
+    if (this._promiseProxy) {
+      if (this._promiseProxy.get('isPending')) {
+        return this._promiseProxy;
       }
     }
 
-    this.setRelationshipIsStale(true);
+    this.setHasFailedLoadAttempt(false);
+    this.setShouldForceReload(true);
```

Comment:
Whats the thinking behind setting this on the object/interaction with the scheduleFetchMany?